src/regression/theil-sen-regression.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+const BaseRegression = require('./base-regression');
+const maybeToPrecision = require('./util').maybeToPrecision;
+
+/**
+ * Theil–Sen estimator
+ *
+ * https://en.wikipedia.org/wiki/Theil%E2%80%93Sen_estimator
+ * @class
+ */
+class TheilSenRegression extends BaseRegression {
+
+    /**
+     *
+     * @param x
+     * @param y
+     * @param options
+     * @constructor
+     */
+    constructor(x, y, options) {
+        options = options || {};
+        super();
+        if (x === true) {
+            // loads the model
+            this.slope = y.slope;
+            this.intercept = y.intercept;
+            this.quality = y.quality || {};
+            if (y.quality.r) {
+                this.quality.r = y.quality.r;
+                this.quality.r2 = y.quality.r2;
+            }
+            if (y.quality.chi2) {
+                this.quality.chi2 = y.quality.chi2;
+            }
+        } else {
+            // creates the model
+            let len = x.length;
+            if (len !== y.length) {
+                throw new RangeError('Input and output array have a different length');
+            }
+
+            let slopes = new Array(len * len);
+            let count = 0;
+            for (let i = 0; i < len; ++i) {
+                for (let j =  i + 1; j < len; ++j) {
+                    if (x[i] !== x[j]) {
+                        slopes[count++] = (y[j] - y[i]) / (x[j] - x[i]);
+                    }
+                }
+            }
+            slopes.length = count;
+            let medianSlope = slopes.sort()[Math.floor(count / 2)];
```

Comment:
This is probably not sorting as you want. The default sort transforms the values to string and uses alphabetical order.
You can use the median from `ml-stat` to get the correct value